chocolatey-diff/Public/Get-FullPackageInfo.ps1

```diff
@@ -19,24 +19,30 @@ function Get-FullPackageInfo {
         [string] $PackageId
     )
 
-    $pkgInfo = @{
-        Id       = $PackageId
-        Url      = "$(Get-ChocolateyPackageGalleryUrl)/${$PackageId}"
-        Versions = @()
-    }
+    if (-not $script:pkgInfo) {
```

Comment:
Won't this have a side-affect that between every time different packages needs to be compared, the user will have to re-import the initial module?

For instance when someone call initially `Get-ChocolateyPackageDiff packageA`

And then afterwards call `Get-ChocolateyPackageDiff packageB` they will get the package information being compared for `packageA`?

Reply:
was about to say something similar: the "cache" really only looks if a given variable exists in script scope, not if the package Id matches.
I'd suggest implementing a "cache map", something like 
```
$script:pkgInfoCache = @{}
```

and then always look if 
```

Reply:
@AdmiringWorm @mwallner 

> Script: The scope that is created while a script file runs. Only the commands in the script run in the script scope. To the commands in a script, the script scope is the local scope.

Source: https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.core/about/about_scopes?view=powershell-7

I just tested it, tested the package `lastpass` and then the package `grafana` I got a normal result

Reply:
... "u sure 'bout that"?
![image](https://user-images.githubusercontent.com/5354972/95782462-3b25dc80-0cd0-11eb-95a3-85e8b17e31a4.png)

Reply:
I am now explicitly checking if the id of the stored data matches the id of the package requested.

Reply:
I'd really like the cache to be more verbose about what it does, and if possible - explicitly differentiate between packageIds.

suggestion:
```
[CmdletBinding()]
    param (
        [Parameter()]
        [string] $PackageId
    )

    if (-not $script:pkgInfoCache) {
        Write-Verbose "initializing package info cache"
        $script:pkgInfoCache = @{}
    }

    if ($script:pkgInfoCache[$PackageId]) {
        Write-Verbose "returning cached result for package '$($PackageId)'"
        Write-Output $script:pkgInfoCache[$PackageId]
        return
    }
    $pkgInfo = @{
        Id       = $PackageId
        Url      = "$(Get-ChocolateyPackageGalleryUrl)/${$PackageId}"
        Versions = @()
    }

    $url = "$(Get-ChocolateyRepositoryUrl)/package-versions/${PackageId}?includePreRelease=true"
    [array]$jsonVersions = Invoke-RestMethod -Uri $url -UseBasicParsing | Select-Object -Unique
    $apiVersions = $jsonVersions |
        ForEach-Object { [SemanticVersion]::create($_) } |
        Sort-Object -Property VersionOnly, Tag -Descending |
        Select-Object -Unique

    foreach ($v in $apiVersions) {
        $vInfo = Get-VersionData -packageName $PackageId -version $v
        $pkgInfo.Versions += @{
            Version = $vInfo.version
            Listed  = $vInfo.listed
            Status  = $vInfo.status
        }
    }

    Write-Verbose "storing result for '$($PackageId)' in package info cache"
    $script:pkgInfoCache[$PackageId] = $pkgInfo
    Write-Output $pkgInfo
```

Reply:
I am only storing one package data result.

Reply:
To save memory, I rather not store more than 1 package in the cache. While I think it might not get that big, when keeping the window open it will not clear automatically.

while that could be solved by a simple script, we need to introduce an extra commandlet / parameter that I don't like.